config/rest_config.go

```diff
@@ -46,12 +59,94 @@ func (c *RestConsumerConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	return nil
 }
 
+
+func (c *RestConsumerConfig) Init(rc *RootConfig) error {
+	restConsumerConfig := rc.RestConsumer
+
+	if restConsumerConfig == nil {
+		logger.Debugf("[rest-protocol] no consumer config")
+		return nil
+	}
+
+	restConsumerServiceConfigMap := make(map[string]*RestServiceConfig, len(restConsumerConfig.RestServiceConfigsMap))
+	for key, rc := range restConsumerConfig.RestServiceConfigsMap {
+		rc.Client = common.GetNotEmptyStr(rc.Client, restConsumerConfig.Client, constant.DefaultRestClient)
+		rc.RestMethodConfigs = initMethodConfigMap(rc, restConsumerConfig.Consumes, restConsumerConfig.Produces)
+		restConsumerServiceConfigMap[key] = rc
+	}
+
+	SetRestConsumerServiceConfigMap(restConsumerServiceConfigMap)
+
+	return nil
+}
+
+// initProviderRestConfig ...
+func initMethodConfigMap(rc *RestServiceConfig, consumes string, produces string) map[string]*RestMethodConfig {
+	mcm := make(map[string]*RestMethodConfig, len(rc.RestMethodConfigs))
+	for _, mc := range rc.RestMethodConfigs {
+		mc.InterfaceName = rc.InterfaceName
+		mc.Path = rc.Path + mc.Path
+		mc.Consumes = common.GetNotEmptyStr(mc.Consumes, rc.Consumes, consumes)
+		mc.Produces = common.GetNotEmptyStr(mc.Produces, rc.Produces, produces)
+		mc.MethodType = common.GetNotEmptyStr(mc.MethodType, rc.MethodType)
+		mc = transformMethodConfig(mc)
+		mcm[mc.MethodName] = mc
+	}
+	return mcm
+}
+
+// transformMethodConfig
+func transformMethodConfig(methodConfig *RestMethodConfig) *RestMethodConfig {
```

Comment:
The methodConfig is a type of pointer, why do you return a pointer again?

Reply:
Because there is an object creation of RestMethodConfigs in the initMethodConfigMap method in rest_config.go In order to reduce the impact of object copying, a pointer is added

Reply:
Yes, I am saying that the **return value** could be removed, cuz the type of the `methodConfig` on the parameter is a pointer.

Reply:
The return value cannot be deleted because RestServiceConfig needs to declare the pointer type RestMethodConfigs map[string]*RestMethodConfig

Reply:
Gotcha!